CMakeLists.txt

```diff
@@ -15,6 +15,10 @@ project(opentelemetry-cpp)
 # Mark variables as used so cmake doesn't complain about them
 mark_as_advanced(CMAKE_TOOLCHAIN_FILE)
 
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.5")
+  set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
Do we need add this by default?I think IDEs should add this option automatically when them need it.(When we use vscode and set C/C++ configurationProvider to `ms-vscode.cmake-tools`, this option will be set by `cmake-tools` plugin.)

And most time when building a package or run in CI, we do not need it.

Reply:
I'm not opposed to putting this behind a CMake flag, but turning it "ON" by default seems the right thing to do from a developer perspective. For example, consider vim or emacs users (personally, I use `neovim`) — in those cases, I'm not aware of any launcher script or anything that sets this in the user profile side of things to propagate to the CMake configure process.

I agree that this flag isn't useful in CI on its own currently; it's mainly targeting local developers. Is there any noticeable "slowdowns" on CI to make this something we should hide behind a flag?

Reply:
I didn't test the cost on otel-cpp yet, but I think it's not a huge project so it may  not cost too much. What's your think about it @lalitb ?

Reply:
I wonder why you don't just use `cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=1`, when configuring?

Reply:
I'm OK setting `-DCMAKE_EXPORT_COMPILE_COMMANDS` when configuring `CMake`.  While this is generally useful to more developers, it doesn't seem there is much interest in having it be set at the top-level, so I'll keep it to my own configuration then. Going to close this PR as such — thanks for the discussion.